### ISSUE TITLE
UniLinksPlugin.h:1:9: fatal error: 'Flutter/Flutter.h' file not found

### ISSUE THREAD

**Issue by marslannasr7koncepts** [avioli/uni_links#118]
My Flutter doctor report is.

Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 2.2.0, on macOS 11.2.3 20D91 darwin-x64, locale
    en-AU)
[!] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    ! Some Android licenses not accepted.  To resolve this, run: flutter doctor
      --android-licenses
[✓] Xcode - develop for iOS and macOS
[✓] Chrome - develop for the web
[✓] Android Studio (version 4.2)
[✓] Connected device (2 available)

! Doctor found issues in 1 category.

**Comment by Andranjo97** [comment]
Have you tried this?
https://github.com/flutter/flutter/issues/70895#issuecomment-744734693

**Comment by ouija** [comment]
> Have you tried this?
> [flutter/flutter#70895 (comment)](https://github.com/flutter/flutter/issues/70895#issuecomment-744734693)

I'm having same issue and the fix via the link doesn't help.
This seems to be a result of having to address the `IPHONEOS_DEPLOYMENT_TARGET` issue by editing the `./ios/Podfile` and changing the `post_install` method from

```
post_install do |installer|
  installer.pods_project.targets.each do |target| 
    flutter_additional_ios_build_settings(target)
  end
end
```
TO:
```
post_install do |installer|
  installer.pods_project.targets.each do |target|
    target.build_configurations.each do |config|
      config.build_settings['ENABLE_BITCODE'] = 'NO'
      config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'] = '10.0'
    end
  end
end
```

And switching this back fixes the issue, however then I'm unable to run the app in the simulator without getting the `The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 14.3.99` error.

Any suggestions?  This was working a few months ago..

**Comment by marslannasr7koncepts** [comment]
Thanks for reply I have fix this issue.

> > Have you tried this?
> > [flutter/flutter#70895 (comment)](https://github.com/flutter/flutter/issues/70895#issuecomment-744734693)
> 
> I'm having same issue and the fix via the link doesn't help.
> This seems to be a result of having to address the `IPHONEOS_DEPLOYMENT_TARGET` issue by editing the `./ios/Podfile` and changing the `post_install` method from
> 
> ```
> post_install do |installer|
>   installer.pods_project.targets.each do |target| 
>     flutter_additional_ios_build_settings(target)
>   end
> end
> ```
> 
> TO:
> 
> ```
> post_install do |installer|
>   installer.pods_project.targets.each do |target|
>     target.build_configurations.each do |config|
>       config.build_settings['ENABLE_BITCODE'] = 'NO'
>       config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'] = '10.0'
>     end
>   end
> end
> ```
> 
> And switching this back fixes the issue, however then I'm unable to run the app in the simulator without getting the `The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 14.3.99` error.
> 
> Any suggestions? This was working a few months ago..

**Comment by ouija** [comment]
> Thanks for reply I have fix this issue.

Did you fix it by changing the `post_install` method in the `./ios/Podfile`?
Are you able to run your application in the iOS simulator?  If so, could you please share your fix?

**Comment by marslannasr7koncepts** [comment]
No Not this way I am adding a line in podfile where adding check related to Simulator version 8.

My Code that I added then this issue resolved is:

**Adding this Line resolve my issue.**
  **flutter_additional_ios_build_settings(target)**


`


post_install do |installer|      installer.pods_project.targets.each do |target|        flutter_additional_ios_build_settings(target)        target.build_configurations.each do |config|          config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'] = '9.0'        end      endend

`

> > Thanks for reply I have fix this issue.
> 
> Did you fix it by changing the `post_install` method in the `./ios/Podfile`?
> Are you able to run your application in the iOS simulator? If so, could you please share your fix?

**Comment by marslannasr7koncepts** [comment]
Yes It's Installed on Simulator Alhumdulilha after putting setting line within pod file where Simulator check added.

> > Thanks for reply I have fix this issue.
> 
> Did you fix it by changing the `post_install` method in the `./ios/Podfile`?
> Are you able to run your application in the iOS simulator? If so, could you please share your fix?

**Comment by ouija** [comment]
Ok, that helped give me some insight.   I ran into some additional issues due to using OneSignal for push notifications _(which requires a minimum iOS target of 10, but was throwing a `building for iOS Simulator, but linking in object file built for iOS` error),_ and managed to resolve [this](https://github.com/OneSignal/OneSignal-iOS-SDK/issues/744#issuecomment-783609842) as well.   For anyone else running into similar issues, my updated `./ios/Podfile` now looks like:

```
target 'OneSignalNotificationServiceExtension' do
  use_frameworks!
  pod 'OneSignalXCFramework', '3.2.1'
end

post_install do |installer|
  installer.pods_project.targets.each do |target|
    flutter_additional_ios_build_settings(target)
    target.build_configurations.each do |config|
        config.build_settings['ENABLE_BITCODE'] = 'NO'
        config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'] = '10.0'
    end
  end
```

**Comment by ouija** [comment]
There is also an [alternative method](https://stackoverflow.com/a/67781691/1330629) that can be used so that you are able to define the mimimal deployment target via the `platform` variable defined within the `Podfile`:
```
post_install do |installer|
  puts 'Determining pod project minimal deployment target'
  pods_project = installer.pods_project
  deployment_target_key = 'IPHONEOS_DEPLOYMENT_TARGET'
  deployment_targets = pods_project.build_configurations.map{ |config| config.build_settings[deployment_target_key] }
  minimal_deployment_target = deployment_targets.min_by{ |version| Gem::Version.new(version) }
  puts 'Minimal deployment target is ' + minimal_deployment_target
  puts 'Setting each pod deployment target to ' + minimal_deployment_target
  installer.pods_project.targets.each do |target|
    flutter_additional_ios_build_settings(target)
    target.build_configurations.each do |config|
      config.build_settings[deployment_target_key] = minimal_deployment_target
      config.build_settings['ENABLE_BITCODE'] = 'NO'
    end
  end
end
```

And then just uncomment the second line at the top of the `Podfile` and replace with your deployment target version:
```
# Uncomment this line to define a global platform for your project
platform :ios, '10.0'
```
